cmd/rerunfails.go

```diff
@@ -83,6 +108,25 @@ func rerunFailed(ctx context.Context, opts *options, scanConfig testjson.ScanCon
 			if exitErr != nil {
 				nextRec.lastErr = exitErr
 			}
+
+			// Need to wait for the go test command to finish before combining
+			// the coverprofile files. But need to before checking for errors.
+			// Even if there is errors, we still need to combine the
+			// coverprofile files.
+			if isCoverprofile {
+				if err := coverprofile.Combine(mainProfile, newCoverprofile); err != nil {
```

Comment:
It might be better to wait until all the attempts are complete and then combine all the files. That way we aren't re-reading the main file each time.

 Would that work?

Reply:
Yes, this would work and better.

Reply:
After some more thoughts, I implemented the logic as follow:

- each rerun will write the coverage profile into it's own file
- after each rerun, read the rerun coverage profile into memory and delete the file
- after all rerun is done before success, we combine all the rerun coverage profile with the main profile

This achieve 2 goals:
- Only reading the main coverage profile once.
- The coverage is valid upto the latest failed rerun and the rerun cover profile file is cleaned up

PTAL and let me know what you think.